### PR TITLE
net-misc/frr: Bump to 7.5.1

### PR DESCRIPTION
**Pull request by jsmolic**
Apart from python update, mostly containes cosmetic changes compared to 7.5

```diff
--- frr-7.5.ebuild
+++ frr-7.5.1.ebuild
@@ -1,21 +1,25 @@
-# Copyright 2020 Gentoo Authors
+# Copyright 2020-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2

 EAPI=7

-PYTHON_COMPAT=( python3_{7,8} )
+PYTHON_COMPAT=( python3_{7..9} )
 inherit autotools pam python-single-r1 systemd

 DESCRIPTION="The FRRouting Protocol Suite"
 HOMEPAGE="https://frrouting.org/"
 SRC_URI="https://github.com/FRRouting/frr/archive/${P}.tar.gz"
+# FRR tarballs have weird format.
+S="${WORKDIR}/frr-${P}"

 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-
 IUSE="doc fpm grpc ipv6 kernel_linux nhrp ospfapi pam rpki snmp systemd test"

+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
 COMMON_DEPEND="
 	${PYTHON_DEPS}
 	acct-user/frr
@@ -27,36 +31,23 @@
 	nhrp? ( net-dns/c-ares:0= )
 	pam? ( sys-libs/pam )
 	rpki? ( >=net-libs/rtrlib-0.6.3[ssh] )
-	snmp? ( net-analyzer/net-snmp )
-"
+	snmp? ( net-analyzer/net-snmp )"

 BDEPEND="
-	doc? ( dev-python/sphinx )
 	sys-devel/flex
 	virtual/yacc
-"
+	doc? ( dev-python/sphinx )"

 DEPEND="
 	${COMMON_DEPEND}
-	test? ( $(python_gen_cond_dep 'dev-python/pytest[${PYTHON_USEDEP}]') )
-"
+	test? ( $(python_gen_cond_dep 'dev-python/pytest[${PYTHON_USEDEP}]') )"

 RDEPEND="
 	${COMMON_DEPEND}
 	$(python_gen_cond_dep 'dev-python/ipaddr[${PYTHON_USEDEP}]')
-	!!net-misc/quagga
-"
-
-PATCHES=(
-	"${FILESDIR}/${PN}-7.5-ipctl-forwarding.patch"
-)
+	!!net-misc/quagga"

-REQUIRED_USE="${PYTHON_REQUIRED_USE}"
-
-RESTRICT="!test? ( test )"
-
-# FRR tarballs have weird format.
-S="${WORKDIR}/frr-${P}"
+PATCHES=( "${FILESDIR}"/${PN}-7.5-ipctl-forwarding.patch )

 src_prepare() {
 	default
@@ -100,12 +91,12 @@
 src_compile() {
 	default

-	use doc && (cd doc; make html)
+	use doc && emake -C doc html
 }

 src_install() {
 	default
-	find "${D}" -name '*.la' -delete || die
+	find "${ED}" -name '*.la' -delete || die

 	# Install user documentation if asked
 	use doc && dodoc -r doc/user/_build/html
@@ -139,12 +130,12 @@
 	newins redhat/frr.logrotate frr

 	# Install PAM configuration file
-	use pam && newpamd "${FILESDIR}/frr.pam" frr
+	use pam && newpamd "${FILESDIR}"/frr.pam frr

 	# Install init scripts
 	systemd_dounit tools/frr.service
-	newinitd "${FILESDIR}/frr-openrc-v1" frr
+	newinitd "${FILESDIR}"/frr-openrc-v1 frr

 	# Conflict files, installed by net-libs/libsmi, bug #758383
-	rm "${D}/usr/share/yang/ietf-interfaces.yang" || die
+	rm "${ED}"/usr/share/yang/ietf-interfaces.yang || die
 }
```

### CODE REVIEW

File: net-misc/frr/frr-7.5.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+HOMEPAGE="https://frrouting.org/"
+SRC_URI="https://github.com/FRRouting/frr/archive/${P}.tar.gz"
+# FRR tarballs have weird format.
+S="${WORKDIR}/frr-${P}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc fpm grpc ipv6 kernel_linux nhrp ospfapi pam rpki snmp systemd test"
+
```
Comment (SoapGentoo) [inline net-misc/frr/frr-7.5.1.ebuild:19]: remove empty line (this is one block in `skel.ebuild`)

File: net-misc/frr/frr-7.5.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	doc? ( dev-python/sphinx )"
+
+DEPEND="
+	${COMMON_DEPEND}
+	test? ( $(python_gen_cond_dep 'dev-python/pytest[${PYTHON_USEDEP}]') )"
+
+RDEPEND="
+	${COMMON_DEPEND}
+	$(python_gen_cond_dep 'dev-python/ipaddr[${PYTHON_USEDEP}]')
+	!!net-misc/quagga"
```
Comment (SoapGentoo) [inline net-misc/frr/frr-7.5.1.ebuild:48]: in general, `!!` have no special bearing in `RDEPEND`. Is this purely for file collisions? If so, make it just `!`

File: net-misc/frr/frr-7.5.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		--disable-static \
+		--with-pkg-extra-version="-gentoo" \
+		--enable-configfile-mask=0640 \
+		--enable-logfile-mask=0640 \
+		--prefix=/usr \
```
Comment (SoapGentoo) [inline net-misc/frr/frr-7.5.1.ebuild:65]: prefix support?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: net-misc/frr/frr-7.5.1.ebuild (hunk 2 above)
Comment (jsmolic) [reply net-misc/frr/frr-7.5.1.ebuild:48]: yeah, I believe it's just for file collisions, I'll remove that one `!`

File: net-misc/frr/frr-7.5.1.ebuild (hunk 1 above)
Comment (jsmolic) [reply net-misc/frr/frr-7.5.1.ebuild]: done

File: net-misc/frr/frr-7.5.1.ebuild (hunk 3 above)
Comment (jsmolic) [reply net-misc/frr/frr-7.5.1.ebuild]: right, done

**Review by SoapGentoo — APPROVED** [review]

**Pull request closed without merging**